docs/source/RunningInEmulationTCG.rst

```diff
@@ -0,0 +1,92 @@
+
+Or you can run any of the ``unattended_install`` tests with
+``--vt-guest-os`` of your choice (very slow).
+
+When running the tests, on top of the usual arguments, you usually
+need to specify:
+
+* ``--vt-qemu-bin`` - path to binary that supports expected architecture
+  (eg. ``/usr/local/bin/qemu-system-aarch64``)
+* ``--vt-qemu-dst-bin`` - for migration tests you need also to specify
+  the destination qemu binary (otherwise default one is used)
+* ``enable_kvm=no`` - to disable KVM (if necessary)
+
+
+aarch64
+=======
+
+ARM always requires `cpu_model` as well as `machine_type`. To
+get list of available models you can run ``qemu-system-aarch64
+-cpu help -M virt`` (note: not all listed cpus are bootable).
+ARM also requires GIC version to be specified. Avocado-vt
+sets automatically the best via ``gic=host``, but that is not
+possible without KVM so you need to override the Avocado-vt
+default by overriding ``machine_type_extra_params``. Complete
+example would be::
+
+   $ avocado vt-bootstrap --vt-guest-os JeOS.27.aarch64
+   $ avocado --show all run --vt-extra-params enable_kvm=no cpu_model=cortex-a57 \\
+       machine_type_extra_params='' --vt-machine-type aarch64 --vt-arch arm64-pci \\
+       --vt-qemu-bin /usr/local/bin/qemu-system-aarch64 -- boot
+
+
+ppc64/ppc64le
+=============
+
+PowerPC can use either BE or LE instructions, but from qemu point of view
+nothing changes. Still for Avocado-vt you either have to specify
+``--vt-arch ppc64`` or ``--vt-arch ppc64le`` to chose the right distribution
+image (both were available as JeOS when writing this document). Apart from
+this no additional tweaks are necessary::
+
+   $ avocado vt-bootstrap --vt-guest-os JeOS.27.ppc64
+   $ avocado --show all run --vt-extra-params enable_kvm=no --vt-arch ppc64 \\
+       --vt-machine-type pseries --vt-qemu-bin /usr/local/bin/qemu-system-ppc64 -- boot
+
+   $ avocado vt-bootstrap --vt-guest-os JeOS.27.ppc64le
+   $ avocado --show all run --vt-extra-params enable_kvm=no --vt-arch ppc64le \\
+       --vt-machine-type pseries --vt-qemu-bin /usr/local/bin/qemu-system-ppc64 -- boot
+
+
+s390x
+=====
+
+For KVM execution Avocado-vt uses ``-cpu host`` on s390x, which is not
+possible without KVM. To execute in TCG mode you need to replace it with
+either a supported CPU type or simply leave it blank::
+
+   $ avocado vt-bootstrap --vt-guest-os JeOS.27.s390x
+   $ avocado --show all run --vt-extra-params enable_kvm=no cpu_model='' --vt-arch s390x \\
+       --vt-machine-type s390-virtio --vt-qemu-bin /usr/local/bin/qemu-system-s390x -- boot
+
+riscv64
+=======
+
+When writing this document, riscv64 was not available as JeOS and even
+Fedora support was not straight forward. See `riscv64_setup`_ for setup
+instructions. Apart from the setup running riscv64 does not require any
+additional arguments::
+
+   $ avocado run --vt-machine-type riscv64-mmio --vt-arch riscv64 \\
+       --vt-extra-params enable_kvm=no --vt-guest-os Fedora.28 -- boot``
```

Comment:
`--vt-guest-os Fedora.28.riscv64`

Reply:
This should not be needed as there is currently only riscv64 version supported by riscv64-mmio (I prefer not to filter things unless necessary as potentially you might leave-out some useful variants).

Reply:
Well uniform to what? I can see 6 occurrences of `--vt-guest-os` in the docs:

```
docs/source/Experimental.rst:         --vt-extra-params enable_kvm=no --vt-guest-os Fedora.28 -- boot``
docs/source/GetStartedGuide.rst:   ``xz`` you can use ``avocado vt-bootstrap --vt-type qemu --vt-guest-os
docs/source/ListingGuests.rst:    $ avocado list --vt-type qemu --vt-guest-os Linux.CentOS.6.6.i386.i440fx --verbose
docs/source/ListingGuests.rst:    $ avocado run balloon_check --vt-type qemu --vt-guest-os Fedora.21
docs/source/WritingTests/DefiningNewGuests.rst:    $ avocado run migrate..tcp --vt-type qemu --vt-guest-os LinuxCustom.FooLinux
docs/source/WritingTests/DefiningNewGuests.rst:    $ avocado run migrate..tcp --vt-type qemu --vt-guest-os WindowsCustom.FooWindows
```

so 4x without the arch, 1x mentioning you should use `OS_OF_YOUR_CHOICE` (not arch, just os) and 1x with arch and machine-type.

In my testing I never specify guest os with the architecture as we have `--vt-machine-type` and `--vt-arch` arguments to do that so I don't think it belongs to documentation (I mean if people are to set architecture from guest-os than we better remove the `--vt-arch` but I'd prefer not to...)

Reply:
btw setting `--vt-arch` and `--vt-guest-os ...arch` might even clash if different archs are provided... I don't know who started to use it, but I don't like it.

Reply:
I also think it's better to make them consistent. Does the `<os>.<version>` format always work? If not (I guess so), the `<os>.<version>.<arch>` format looks better to me.

BTW, the actual file on my test system is jeos-27-aarch64.qcow2, not jeos.27.aarch64. So avocado-vt replaces all "." with "-" to get the actual file name?

Reply:
Thanks for your explanation. If I understand it correctly, you meant values of `--vt-guest-os`, `--vt-arch` and `--vt-machine-type` options are of `<filter rule1>.<filter rule2>...` format and user are responsible to put a meaningful rule for each option, right? I have no more comments on this.

Reply:
Yes, basically they are there to simplify the access but in reality it's basically just a symlink to `--vt-only-filter`.

Reply:
okay, Thanks for explanation. LGTM.